src/Common/AsynchronousMetrics.cpp

```diff
@@ -797,7 +797,7 @@ void AsynchronousMetrics::update(TimePoint update_time)
 
             int64_t hz = sysconf(_SC_CLK_TCK);
             if (-1 == hz)
-                throwFromErrno("Cannot call 'sysconf' to obtain system HZ", ErrorCodes::CANNOT_SYSCONF);
```

Comment:
But then you lose errno. We don't want to lose this detail.

Reply:
It's still being captured

https://github.com/ClickHouse/ClickHouse/pull/57910/files#diff-9884cbf5c6e44559e86a36073ca63b3dd91929be7123373310d6e824af4504a6R190-R193

Reply:
Why do we still need `ErrnoException::throwWithErrno` then?

Reply:
For the case where errno is saved, other ops are done and then the exception is raised. Also when the error code doesn’t come from errno but from the result of an operation. I think there is only a handful of calls